### PR TITLE
Fix L4 proxy post mesh

### PR DESCRIPTION
**Pull request by dndx**

### CODE REVIEW

File: kong/conf_loader.lua [hunk 1]
```diff
@@ -1041,7 +1042,7 @@ local function load(path, custom_conf, opts)
   do
     local http_flags = { "ssl", "http2", "proxy_protocol", "transparent",
                          "deferred", "bind", "reuseport", "backlog=%d+" }
-    local stream_flags = { "proxy_protocol", "transparent", "bind",
+    local stream_flags = { "ssl", "proxy_protocol", "transparent", "bind",
```
Comment (bungle) [inline kong/conf_loader.lua]: Do we still ship `trasnparent` patch? I though it was gone, and thus needs to be removed everywhere?

Comment (bungle) [reply kong/conf_loader.lua]: it is not here anymore: https://github.com/Kong/openresty-patches/tree/fix/remove_mesh/patches/1.15.8.2

Comment (bungle) [reply kong/conf_loader.lua]: I created PR to that:
https://github.com/Kong/kong/pull/5350

Comment (dndx) [reply kong/conf_loader.lua]: I wasn't sure about the use case of transparent proxy first, but if you said so I'm happy to see it got removed :)

File: kong/conf_loader.lua [hunk 2]
```diff
@@ -1007,7 +1007,7 @@ local function load(path, custom_conf, opts)
   do
     local http_flags = { "ssl", "http2", "proxy_protocol", "deferred",
                          "bind", "reuseport", "backlog=%d+" }
-    local stream_flags = { "proxy_protocol", "bind", "reuseport" }
+    local stream_flags = { "ssl", "proxy_protocol", "bind", "reuseport" }
```
Comment (bungle) [inline kong/conf_loader.lua]: `backlog` could be added here too.

File: spec/fixtures/custom_nginx.template [hunk 3]
```diff
@@ -693,37 +693,44 @@ stream {
...
 > for _, el in ipairs(nginx_sproxy_directives) do
-        $(el.name) $(el.value);
+    $(el.name) $(el.value);
 > end
 
-> if ssl_preread_enabled then
-        ssl_preread on;
+> if stream_proxy_ssl_enabled then
+    ssl_certificate ${{SSL_CERT}};
+    ssl_certificate_key ${{SSL_CERT_KEY}};
```
Comment (bungle) [inline spec/fixtures/custom_nginx.template]: Should we have this too:

```suggestion
    ssl_certificate_key ${{SSL_CERT_KEY}};
> if ssl_ciphers then
    ssl_ciphers ${{SSL_CIPHERS}};
> end
```

File: kong/templates/nginx_kong_stream.lua [hunk 4]
```diff
@@ -84,18 +85,24 @@ server {
     $(el.name) $(el.value);
 > end
 
-> if ssl_preread_enabled then
-    ssl_preread on;
+> if stream_proxy_ssl_enabled then
+    ssl_certificate ${{SSL_CERT}};
+    ssl_certificate_key ${{SSL_CERT_KEY}};
```
[the remark this answers is not in the document]
Comment (hishamhm) [reply kong/templates/nginx_kong_stream.lua]: `ssl_ciphers` is not wrapped in `> if` in the main `nginx_kong.lua` template. Can we just add it unconditionally here?

[the remark this answers is not in the document]
Comment (hishamhm) [reply kong/templates/nginx_kong_stream.lua]: Ah, I was looking at the `master` template. In the one in `next` it is wrapped.

**Comment by hishamhm** [comment]
just waiting for a CI run of the latest rebased code

**Pull request merged**